include/colby/conversions.hpp

```diff
@@ -0,0 +1,164 @@
+/**
+ *	\file
+ */
+
+#pragma once
+
+#include <opencv2/core/mat.hpp>
+#include <opencv2/core/matx.hpp>
+#include <iostream>
+#include <cmath>
+
+namespace colby {
+
+inline cv::Vec3f xyz2lab(const cv::Vec3f xyz) noexcept {
+	auto x = xyz[0]/95.047f;
+	auto y = xyz[1]/100.f;
+	auto z = xyz[2]/108.883f;
+
+	x = x > 0.008856f ? std::pow(x, 1.f/3.f) : 7.787f*x+16.f/116.f;
+	y = y > 0.008856f ? std::pow(y, 1.f/3.f) : 7.787f*y+16.f/116.f;
+	z = z > 0.008856f ? std::pow(z, 1.f/3.f) : 7.787f*z+16.f/116.f;
+
+	auto l = 116.f*y-16.f;
+	auto a = 500.f*(x-y);
+	auto b = 200.f*(y-z);
+
+	return cv::Vec3f(l,a,b);
+}
+
+inline cv::Vec3f lab2xyz(const cv::Vec3f lab) noexcept {
+	auto l = lab[0];
+	auto a = lab[1];
+	auto b = lab[2];
+
+	auto y = (l+16.f)/116.f;
+	auto x = a/500.f+y;
+	auto z = y-b/200.f;
+
+	x = std::pow(x, 3.f)>0.008856 ? std::pow(x, 3.f)  : (x-16.f/116.f)/7.787f;
+	y = std::pow(y, 3.f)>0.008856 ? std::pow(y, 3.f)  : (y-16.f/116.f)/7.787f;
+	z = std::pow(z, 3.f)>0.008856 ? std::pow(z, 3.f)  : (z-16.f/116.f)/7.787f;
+
+	x *= 95.047;
+	y *= 100.f;
+	z *= 108.883f;
+
+	return cv::Vec3f(x,y,z);
+}
+
+inline cv::Vec3f bgr2xyz(const cv::Vec3b bgr) noexcept {
+	auto r = bgr[2]/255.f;
+	auto g = bgr[1]/255.f;
+	auto b = bgr[0]/255.f;
+
+	r = r>0.04045f ? std::pow(((r+0.055f)/1.055f), 2.4f) : r/12.92f;
+	g = g>0.04045f ? std::pow(((g+0.055f)/1.055f), 2.4f) : g/12.92f;
+	b = b>0.04045f ? std::pow(((b+0.055f)/1.055f), 2.4f) : b/12.92f;
+
+	r *= 100.f;
+	g *= 100.f;
+	b *= 100.f;
+
+	auto x = r*0.4124f+g*0.3576f+b*0.1805f;
+	auto y = r*0.2126f+g*0.7152f+b*0.0722f;
+	auto z = r*0.0193f+g*0.1192f+b*0.9505f;
+
+	return cv::Vec3f(x,y,z);
+}
+
+inline cv::Vec3b xyz2bgr(const cv::Vec3f xyz) noexcept {
+	auto x = xyz[0]/100.f;
+	auto y = xyz[1]/100.f;
+	auto z = xyz[2]/100.f;
+
+	auto r = x*3.2406f+y*-1.5372f+z*-0.4986f;
+	auto g = x*-0.9689f+y*1.8758f+z*0.0415f;
+	auto b = x*0.0557f+y*-0.2040f+z*1.0570f;
+
+	r = r > 0.0031308f ? 1.055f*std::pow(r, 1.f/2.4f)-0.055f : 12.92f*r;
+	g = g > 0.0031308f ? 1.055f*std::pow(g, 1.f/2.4f)-0.055f : 12.92f*g;
+	b = b > 0.0031308f ? 1.055f*std::pow(b, 1.f/2.4f)-0.055f : 12.92f*b;
+
+	r *= 255.f;
+	g *= 255.f;
+	b *= 255.f;
+
+	if (r < 0.f) r = 0.f;
+	if (g < 0.f) g = 0.f;
+	if (b < 0.f) b = 0.f;
```

Comment:
You can just use 0, 0 has special rules that allow it to be a pointer, integer, and floating point constant